server/app/repository/VersionRepository.java

```diff
@@ -344,6 +344,29 @@ public Version getActiveVersion() {
         .findOne();
   }
 
+  /**
+   * Returns the previous version to the one passed in. If there is only one version there isn't a
+   * previous version so the Optional result will be empty.
+   *
+   * <p>This can return draft, active, or obsolete versions. Versions flagged as deleted are
+   * excluded.
+   */
+  public Optional<Version> getPreviousVersion(Version version) {
```

Comment:
**suggestion (blocking):** please add a unit test in VersionRepository for this

Reply:
done